src/main/java/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -19,8 +20,17 @@ public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificat
     @Query(value = "SELECT COUNT(*) > 0 FROM books WHERE isbn = :isbn", nativeQuery = true)
     Long existsByIsbnIncludingDeleted(@Param("isbn") String isbn);
 
-    @Query(value = "SELECT b FROM Book b JOIN b.categories c WHERE c.id = :categoryId")
-    Page<Book> findAllByCategoryId(@Param("categoryId") Long categoryId, Pageable pageable);
+    @Query("""
+            SELECT new onlinebookstore.dto.book.BookDtoWithoutCategoryIds(
```

Comment:
don't return dto from repository. return book and use mapper on the service layer